controller/lib/core/sensors.cpp

```diff
@@ -158,3 +173,40 @@ VolumetricFlow pressure_delta_to_flow(Pressure delta) {
                          venturiAreaProduct * bernoulliAreaDivisor *
                          sqrtf(abs(delta.kPa()) * 1000.0f));
 }
+
+Volume integrate_flow(VolumetricFlow flow) {
+  // volume integral computation data
+  static Time last_flow_measurement_time = millisSinceStartup(0);
```

Comment:
Ideally this also shouldn't have static state but instead be part of the controller object of which main() has a static instance.